master/getting-started/kubernetes/tutorials/app-layer-policy/index.md

```diff
@@ -0,0 +1,298 @@
+of the following command:
+   
+   ```bash
+   export GATEWAY_URL=$(kubectl get po -n istio-system -l istio=ingress -o 'jsonpath={.items[0].status.hostIP}'):$(kubectl get svc istio-ingress -n istio-system -o 'jsonpath={.spec.ports[0].nodePort}')
+   ```
+
+Point your browser to `http://$GATEWAY_URL/` to confirm the YAO Bank application is functioning 
+correctly.  It may take several minutes for all the services to come up and respond, during which
+time you may see 404 or 500 errors.
+
+### The need for policy
+
+Although Calico & Istio are running in the cluster, we have not defined any authorization policy.  
+Istio was configured to mutually authenticate traffic between the pods in your application, so only 
+connections with Istio-issued certificates are allowed, and all inter-pod traffic is encrypted with 
+TLS.  That's already a big step in the right direction.
+
+But, let's consider some deficiencies in this security architecture:
+
+ * All incoming connections from workloads in the Istio mesh are equally trusted
+ * Possession of a key & certificate pair is the *only* access credential considered.
+ 
+To understand why these might be a problem, let's take them one at a time.
+
+#### Trusting workloads
+
+Trusting connections from any workload in the Istio mesh is a poor security architecture because, 
+like Kubernetes, Istio is designed to host multiple applications.  Some of those applications may 
+not be as trusted as others.  They may be operated by different users or teams with wildly different
+security requirements.  We don't want our secure financial application microservices accessible from
+some hacky prototype another developer is cooking up.
+
+Even within our own application, the best practice is to limit access as much as possible.  Only 
+pods that need access to a service should get it.  Consider the YAO Bank application.  The customer
+web service does not need, and should not have direct access to the backend database.  The customer 
+web service needs to directly interact with clients outside the cluster, some of whom may be
+malicious.  Unfortunately, vulnerabilities in web applications are all too common.  For example, an
+[unpatched vulnerabiltiy in Apache Struts][struts cve] is what allowed attackers their initial
+access into the Equifax network where they then launched a devastating attack to steal millions of
+people's financial information.
+
+Imagine what would happen if an attacker were to gain control of the customer web pod in our
+application.
+
+Let's simulate this by `exec`'ing into that pod.
+
+    kubectl exec -ti customer-<fill in pod ID> -c customer bash
+
+You should get a bash shell inside the customer pod.  Notice that from here, we get direct access to
+the backend database.  For example, we can list all the entries in the database like this:
+
+    curl http://database:2379/v2/keys?recursive=true | python -m json.tool
```

Comment:
I feel we should mention something about it being http (without the **s**) but still is encrypted by the Envoy especially since we do make that point in the following attack scenario.
your call.

Reply:
This is in a description of an attack scenario. Given that we are demonstrating that an attacker is able to dump the contents of a database, mentioning that the connection is encrypted while this happens feels out of place and might blunt the impact.